src/main/java/org/jenkinsci/plugins/configfiles/folder/FolderConfigFileAction.java

```diff
@@ -21,27 +21,57 @@
 import org.jenkinsci.plugins.configfiles.Messages;
 import org.kohsuke.accmod.Restricted;
 import org.kohsuke.accmod.restrictions.NoExternalUse;
-import org.kohsuke.stapler.*;
+import org.kohsuke.stapler.HttpRedirect;
+import org.kohsuke.stapler.HttpResponse;
+import org.kohsuke.stapler.QueryParameter;
+import org.kohsuke.stapler.StaplerProxy;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
 
-import javax.servlet.ServletException;
-import java.io.IOException;
-import java.util.*;
+import hudson.Extension;
+import hudson.Util;
+import hudson.model.Action;
+import hudson.model.Item;
+import hudson.model.Job;
+import hudson.security.ACL;
+import hudson.security.Permission;
+import hudson.util.FormValidation;
+
+import com.cloudbees.hudson.plugins.folder.AbstractFolder;
+
+import jenkins.model.TransientActionFactory;
+import net.sf.json.JSONObject;
 
 public class FolderConfigFileAction implements Action, ConfigFilesUIContract, StaplerProxy {
 
-    private Folder folder;
+    private AbstractFolder<?> folder;
 
-    FolderConfigFileAction(Folder folder) {
+    FolderConfigFileAction(AbstractFolder<?> folder) {
         this.folder = folder;
     }
 
-    public Folder getFolder() {
+    public AbstractFolder<?> getFolder() {
         return folder;
     }
 
     @Override
     public String getIconFileName() {
-        return folder.hasPermission(Item.EXTENDED_READ) ? ConfigFilesManagement.ICON_PATH : null;
+        // whilst you may be able to read configuration of a ComputedFolder they are not configurable, so the link should never be shown.
+        boolean hasPerm = folder.hasPermission(Item.EXTENDED_READ);
```

Comment:
This code does not work....
I wonder if it would be better to only show it when you have `Item.CONFIGURE || (Item.EXTENDED_READ && hasatLeastOneManagedFile()`)